psp/models/recent_history.py

```diff
@@ -125,19 +125,24 @@ def __init__(
         *,
         pv_data_source: PvDataSource,
         nwp_data_source: NwpDataSource | None,
+        excarta_data_source: NwpDataSource | None,
```

Comment:
Similar to above, not sure how easy it would be, but potentially making the `nwp_data_source` be `nwp_data_sources: List[NwpDataSource] | None` so its a bit more flexible without adding new data sources to the init args every time.

Reply:
:+1: Or alternatively a dictionary to give them names (for the feature names later).